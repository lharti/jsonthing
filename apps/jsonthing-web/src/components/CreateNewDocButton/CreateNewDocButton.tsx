'use client'

import { Button, ButtonProps } from '@/components/ui/Button'
import { useCreateDoc } from '@/hooks/useCreateDoc'
import { useRouter } from 'next/navigation'
import React from 'react'

type CreateNewDocButtonProps = ButtonProps

export const CreateNewDocButton: React.FC<CreateNewDocButtonProps> = ({
    ...buttonProps
}) => {
    const { createDoc } = useCreateDoc()

    const router = useRouter()

    return (
        <Button
            onClick={() => {
                createDoc(undefined, {
                    onSuccess: ({ data }) => {
                        router.push(`/docs/${data._id}`)
                    },
                })
            }}
            {...buttonProps}
        >
            {'New Doc'}
        </Button>
    )
}
