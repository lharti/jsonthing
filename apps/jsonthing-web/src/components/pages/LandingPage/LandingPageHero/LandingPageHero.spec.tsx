import { CreateNewDocButton } from '@/components/CreateNewDocButton'
import { render } from '@testing-library/react'
import React from 'react'
import { LandingPageHero } from '.'

jest.mock('@/components/CreateNewDocButton')

describe('<LandingPageHero />', () => {
    it('should render', () => {
        expect.assertions(1)

        jest.mocked(CreateNewDocButton).mockReturnValueOnce(
            <button>{'Create new JSON doc'}</button>,
        )

        const { container } = render(<LandingPageHero />)

        expect(container).toMatchSnapshot('<LandingPageHero />')
    })
})
