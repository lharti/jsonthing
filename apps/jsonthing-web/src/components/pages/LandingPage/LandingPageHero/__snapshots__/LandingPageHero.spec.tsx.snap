// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<LandingPageHero /> should render: <LandingPageHero /> 1`] = `
<div>
  <section
    class="flex flex-col items-center gap-8 lg:flex-row lg:items-start lg:justify-between"
  >
    <div
      class="space-y-6 max-w-96 text-center lg:text-start"
    >
      <a
        aria-label="jsonthing logo"
        class="inline-block rounded-md bg-gradient-to-r from-green-400 to-blue-500 px-3 py-1 text-2xl font-semibold text-white shadow-lg"
        href="/"
      >
        jsonthing
      </a>
      <h1
        aria-label=""
        class="
                  text-4xl font-bold

                  md:text-5xl
                "
      >
        JSON storage that speaks API.
      </h1>
      <p
        class="text-xl text-gray-600"
      >
        Edit, and access your JSON documents with automatically generated API endpoints.
      </p>
      <button>
        Create new JSON doc
      </button>
    </div>
    <img
      alt="jsonthing hero"
      class="
                  mt-0

                  lg:mt-12
                "
      data-nimg="1"
      decoding="async"
      height="600"
      loading="lazy"
      src="https://storage.googleapis.com/jsonthing-assets/hero.png"
      style="color: transparent;"
      width="700"
    />
  </section>
</div>
`;
